iroh-p2p/src/service.rs

```diff
@@ -472,12 +490,31 @@ pub async fn build_transport(local_key: Keypair) -> Boxed<(PeerId, StreamMuxerBo
         core::upgrade::SelectUpgrade::new(yamux_config, mplex_config)
     };
 
-    transport
-        .upgrade(core::upgrade::Version::V1Lazy)
-        .authenticate(auth_config)
-        .multiplex(mplex_config)
-        .timeout(Duration::from_secs(20)) // TODO: configurable
-        .boxed()
+    if config.relay_client {
+        let (relay_transport, relay_client) =
+            libp2p::relay::v2::client::Client::new_transport_and_behaviour(
+                local_key.public().to_peer_id(),
+            );
+
+        let transport = OrTransport::new(relay_transport, transport);
+        let transport = transport
+            .upgrade(core::upgrade::Version::V1Lazy)
+            .authenticate(auth_config)
+            .multiplex(mplex_config)
```

Comment:
Not related to this pull request, though very relevant in general, why use `mplex` and not `yamux` or both with a preference on `yamux`?

https://github.com/libp2p/specs/pull/402

Reply:
if you look a little above it is actually both mplex and yamux just the name is wrong 😅

Reply:
:+1: Please keep prioritizing Yamux over Mplex.